### PR TITLE
NEXT-12632 - Add guide about modifying sitemap entries

### PR DESCRIPTION
**Pull request by PaddyS**
https://github.com/shopware/docs/pull/259 should be merged first!

### CODE REVIEW

File: guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    return new UrlResult($urls, $urlResult->getNextOffset());
+}
+```
+{% endcode %}
+
+You could iterate over the `$urls` array and modify each entry - or even create a new array with less entries,
+if you want to fully remove some.
+
+There is one main downside to this way:
+You don't have access to many information about the entity itself, that was used for this `Url` instance.
```
Comment (lernhart) [inline guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md:117]:
```suggestion
You don't have access to a lot of information about the entity itself, that was used for this `Url` instance.
```

File: guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Also, it's not the best way in terms of performance to read all SEO URLs from the database, only to filter them afterwards.
+
+### Overriding the getSeoUrls method
+
+The available SEO URLs are read in the protected method `getSeoUrls` of the `AbstractUrlProvider`.
+Since it's a protected method, you can override it and create a custom SQL in order to only read the data you really want.
+
+For this you'll most likely want to copy the original method's code and paste it into your overridden method.
+You can then add new lines to the SQL statement in order to do the necessary filtering or customising.
```
Comment (lernhart) [inline guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md:129]:
```suggestion
You can then add new lines to the SQL statement in order to do the necessary filtering or customizing.
```

**Review by lernhart — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        return $this->getDecorated()->getUrls($context, $limit, $offset);
+    }
+}
+```
+{% endcode %}
+{% endtab %}
+
+{% tab title="services.xml" %}
+{% code title="<plugin root>/src/Resources/config/services.xml" %}
+```markup
```
Comment (JoshuaBehrens) [inline guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md:73]: I guess you want to highlight xml, right?

**Review by JoshuaBehrens — CHANGES_REQUESTED** [review]

File: guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md (hunk 3 above)
Comment (PaddyS) [reply guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md:73]: Wait, I'm pretty sure I copied & pasted it. There might be more wrong occurrences, thanks! Will fix them all in an extra commit.

Comment (PaddyS) [reply guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md:73]: Welp... The GitBook linter changed a lot of them to `markup` (including twig, html and xml...)

Comment (JoshuaBehrens) [reply guides/plugins/plugins/content/sitemap/modify-sitemap-entries.md]: still reads weird. Probably other markdown renderers are more verbose

**Pull request merged**